CIPs/CIP-23/CIP-23.md

```diff
@@ -0,0 +1,176 @@
+| -------------- | ------------------------------------------------------------ | ---------------- | --------- | -------- | -------------------------------------------- |
+| `protocol`     | Network protocol used to resolve the host                    | string           | 50 char   | required | https                                        |
+| `host`         | Host used to resolve the ID                                  | string           | 150 char  | required | https://github.com                           |
+| `id`           | Path of the unique ID within the host. Can be the same as the host to represent the root. | string           | 450 char  | required | https://github.com/marysmith                 |
+| `claim`        | Unique location where the DID string is stored. Should be a place where only the given ID could have posted the DID. | string           | 450 char  | optional | https://gist.github.com/marysmith/5c48debdb7 |
+| `attestations` | Attestations issued by third-party services which have verified the claim against the ID.<br />The key in this object should be "did-jwt-vc" in order to indicate a verifiable credential. <br />For backwards compatibility reasons "did-jwt" is also supported. | array of objects | 1000 char | optional | {did-jwt-vc: "eyJhbGciOiJIUzI1..."}          |
+
+**Deployment:** `<reference-schema-DocID>`
+
+```jsx
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "array",
+  "title": "AlsoKnownAs",
+  "items": {
+    "$ref": "#/definitions/Account"
+  },
+  "definitions": {
+    "Attestation": {
+      "type": "object",
+      "properties": {
+        "did-jwt": {
+          "type": "string",
+          "maxLength": 1000
+        },
+        "did-jwt-vc": {
+          "type": "string",
+          "maxLength": 1000
+        }
+      }
+    },
+    "Account": {
+      "type": "object",
+      "properties": {
+        "protocol": {
+          "type": "string",
+          "maxLength": 50
+        },
+        "host": {
+          "type": "string",
+          "maxLength": 150
+        },
+        "id": {
+          "type": "string",
+          "maxLength": 450
+        },
+        "claim": {
+          "type": "string",
+          "maxLength": 450
+        },
+        "attestations": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Attestation"
+          }
+        }
+      },
+      "required": [
+        "protocol",
+        "host",
+        "id"
+      ]
+    }
+  }
+}
+```
+
+### Tags
+
+In the config of the *definition* defined above we can see a `tags` property. We simply apply the tags in the config to the `tags` property in the *metadata* of the *reference* Tile.
+
+### Example
+
+An example Social Accounts *reference* document that includes two accounts, Twitter and Github.
+
+```js
+const profile = await ceramic.createDocument('tile', {
+  metadata: {
+    schema: "<reference-schema-DocID>"
+    tags: ["AlsoKnownAs"]
+  },
+  content: [{
```

Comment:
Putting this as an array means that if someone wants to look up the twitter handle for a given DID, they have to load this entire document and then iterate over this array to find the twitter entry.  Should this instead be a map keyed by 'host' to make it easier to locate specific entries?

Reply:
Last time we discussed this you suggested that we put the `host` as the key and value being an array of accounts. Now I'm leaning towards keeping the current design for two reasons:
1. All protocols might not require a "host" (they only use protocol and id)
2. You would still need to load all data, seems like it wouldn't be too much overhead to just iterate though the array.

Reply:
@oed I think host should be required, in some cases though, host and ID could be the same. This would be the case, for example, if you wanted to verify a root domain. So if this is true, then I think what @stbrody is saying could make sense.

Reply:
But if you are looking for a root domain you would want to filter by protocol (DNS) rather than the actual host so having the host as map doesn't really make sense in that case? Actually unclear what host would be in the case of DNS 🤔

Reply:
Do we expect each type of supported AKA link to require being defined in a CIP, or are we hoping for a generic format that enables arbitrary links we haven't thought about yet?  If the former, we could require each type of link to have a unique identifier for that link type e.g. "twitter", "github", "dns" that could be the key in the map.  If we support arbitrary links though then it's harder to guarantee that everyone uses the same identifier for the same type of link.  Although in that case the structure of the link documents could be arbitrary also, or use slightly different "host" field values, so maybe we're no worse off?

Reply:
I think the original intent behind this was that it should support arbitrary new types of entries. *protocol*, *host*, and *id* is fairly generic and you could likely squeeze most things into it 🤷